Exploration-of-the-Oppia-stack.md

```diff
@@ -0,0 +1,431 @@
+## Table of Contents
+
+* [Support editing](#support-editing)
+  * [Frontend changes to support editing](#frontend-changes-to-support-editing)
+  * [Backend changes to support editing](#backend-changes-to-support-editing)
+* [Support viewing](#support-viewing)
+  * [Backend changes to support viewing](#backend-changes-to-support-viewing)
+  * [Frontend changes to support viewing](#frontend-changes-to-support-viewing)
+
+To better understand how Oppia works, let's work through a simple full-stack change--adding a field to a user's preferences. We'll discuss how to update the frontend code to display the new field and how to make the backend store this new field in the database.
+
+Before working through this tutorial, you should read through the [[Overview of Oppia's codebase|Overview-of-the-Oppia-codebase]].
+
+**This tutorial is designed to help you understand how Oppia works, but it is not a template for how you should actually make a code change.** For example, we won't be writing any tests or design docs, which are essential parts of any real full-stack change.
```

Comment:
From @seanlip: Explain the big picture, which is that when making a code change, it can be hard to figure out what files we need to change. The goal for this wiki page is to help readers learn how to find the relevant files.

1. Problem statement: Suppose we are trying to cultivate a learning community among Oppia learners. To help with that, we want to add a field to user profiles where users can describe what fields they have experience with and offer to help people with questions about those fields.
2. Walk through identifying the files that control the user profile and adding code to confirm that we have identified all of them.
3. At the end we'll have a functioning profile field, but this doesn't mean you're ready to open a PR. It just means you've figured out what files you need to change. Now that you know how to build your new feature, you need to open a design doc to get feedback on your approach. You should also:
   * Consider data privacy, ethics, and migration in your design doc
   * Use stacked PRs
   * Write tests
   * Update documentation
   * Consult UI experts

Reply:
done